### PR TITLE
Add basic image io and remove python C-API refs from numpy_returns.cpp

### PR DESCRIPTION
**Pull request by visionworkz**
This is a consolidation of the discussion in https://github.com/davisking/dlib/pull/1249

For your review @davisking and @supervacuus 

@davisking I tried to fix the build but it cannot load the image in the test folder. I verified that the image exists. Maybe an issue related to DLIB_JPEG_SUPPORT flag?

There is further impact due to https://github.com/davisking/dlib/issues/1252

@aldanor if this PR is merged before your PR, you might wish to `ifdef` some portions of the code in this PR as well.

### CODE REVIEW

**Comment by aldanor** [comment]
@visionworkz I started a local branch (with ifdef stuff) this weekend but haven't finished it yet. I'll wait for this batch of work to get merged in first then.

**Comment by davisking** [comment]
@visionworkz, all the file paths when running `python setup.py test` should be relative to the root of the repository.  The ones in the new tests are relative to the python/test folder, I'm pretty sure that's why you are getting the file load errors.

File: tools/python/test/test_numpy_returns.py [hunk 1]
```diff
@@ -24,17 +34,26 @@ def get_test_face_chips():
 def get_test_face_chip():
     rgb_img, shape = get_test_image_and_shape()
     return dlib.get_face_chip(rgb_img, shape)
- 
+
+# The tests below will be skipped for Python 2.7 as there is a pickle issue
```
Comment (davisking) [inline tools/python/test/test_numpy_returns.py]: This should work in python 2.7.  Can't you pickle using an earlier pickling version, like 2? It's certainly not the case that pybind11 pickling doesn't work in python 2.7.

Comment (davisking) [reply tools/python/test/test_numpy_returns.py]: If you use cPickle and a call like .dumps(variable,2) it works.  See http://pybind11.readthedocs.io/en/stable/advanced/classes.html#pickling-support

**Comment by davisking** [comment]
Looks awesome, thanks for the nice PR :)

**Pull request merged**